packages/tgpu-dev-cli/verify-publish-tag.mjs

```diff
@@ -0,0 +1,33 @@
+// @ts-check
+
+import { type } from 'arktype';
+import { readPackageJSON } from 'pkg-types';
+
+const PublishTag = type('"alpha" | "beta" | undefined');
+
+const packageJSON = await readPackageJSON();
+const chosenPublishTag = PublishTag.assert(process.env.npm_config_tag);
+
+export function verifyPublishTag() {
+  let tagVerified = false;
+  for (const { unit: tag } of /** @type {{ unit: string }[]} */ (
+    PublishTag.json
+  )) {
+    if (packageJSON.version?.includes(tag)) {
+      if (tag !== chosenPublishTag) {
+        throw new Error(
+          `Publishing under a mismatched tag "${chosenPublishTag}" for version ${packageJSON.version}. Use --tag ${tag}`,
+        );
+      }
+
+      tagVerified = true;
+      break;
+    }
+  }
+
+  if (!tagVerified) {
+    throw new Error(
+      `Publishing under a mismatched tag "${chosenPublishTag}" for version ${packageJSON.version}.`,
```

Comment:
If I'm not mistaken if there's no tag provided (it's undefined) then this will always throw. and e.g. for `typegpu` package we want there to be no tag. so I think the logic for undefined needs to be changed, we cannot look for "undefined" in version number